custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/scala/com/datarobot/custom/H2OPredictor.scala

```diff
@@ -0,0 +1,199 @@
+      .setConvertInvalidNumbersToNa(true)
+  }
+
+  // this would work with a properly compiled / packaged pojo
+  // this will work with a compile pojo
+  // the expectation is that we compile the pojo when the container is spun up.
+// s"javac -cp code/h2o-genmodel-3.30.0.6.jar ${pojo.getAbsolutePath}".!
+
+  def modelConfigViaPojo(
+      pojo: java.io.File
+  ): EasyPredictModelWrapper.Config = {
+    val urls = pojo.getName.endsWith("java") match {
+      case true => {
+        val pathOfJar = classOf[hex.genmodel.GenModel].getProtectionDomain.getCodeSource.getLocation.toURI.getPath
+        s"javac -cp ${pathOfJar} ${pojo.getAbsolutePath}".!
+        Array(pojo.getAbsoluteFile.getParentFile.toURI.toURL)
+      }
+      case false => throw new Exception("Pojo not availabe")
+    }
+
+    val pojoName = pojo.getName.replace(".jar", "").replace(".java", "")
+    val urlClassLoader = URLClassLoader.newInstance(
+      urls,
+      Thread.currentThread().getContextClassLoader()
+    )
+    val h2oRawPredictor =
+      urlClassLoader.loadClass(pojoName).newInstance.asInstanceOf[GenModel]
+    val modelConfig = new EasyPredictModelWrapper.Config()
+      .setModel(h2oRawPredictor)
+      .setConvertUnknownCategoricalLevelsToNa(true)
+      .setConvertInvalidNumbersToNa(true)
+    // EasyPredictModelWrapper = new EasyPredictModelWrapper(modelConfig)
+    modelConfig
+  }
+
+  def loadModel(modelDir: String): EasyPredictModelWrapper = {
+
+    // h2o model artifacts
+    val re = new scala.util.matching.Regex("(.java$)|(.zip$)")
+
+    val files = Paths.get(modelDir).toFile().listFiles().filter { f =>
+      re.findAllMatchIn(f.getName).hasNext
+    }
+
+    val file = files.length match {
+      case 0 =>
+        throw new Exception("no model artifact found in model directory")
+      case 1 => files.apply(0)
+      case _ =>
+        throw new Exception(
+          s"more than 1 main model artifact found in model directory: ${files.map { _.getName }.mkString(",")}"
+        )
+    }
+
+    val fileExt = re.findAllIn(file.getName).next
+
+    val modelConfig = fileExt match {
+      case ".zip" => modelConfigViaMojo(file)
+      case ".java"  => modelConfigViaPojo(file)
+      case _ => throw new Exception("""|no usable H2O model artifact available""".stripMargin)
+    }
+
+    val model = new EasyPredictModelWrapper(modelConfig)
+
+    model
+  }
+
+  def configure(
+      params: java.util.Map[String, Any] = new java.util.HashMap[String, Any]()
+  ) = {
+
+    model = loadModel(
+      params.get("__custom_model_path__").asInstanceOf[String]
+    )
+    customModelPath = params.get("__custom_model_path__").asInstanceOf[String]
+    negativeClassLabel = params.get("negativeClassLabel").asInstanceOf[String]
+    positiveClassLabel = params.get("positiveClassLabel").asInstanceOf[String]
+
+    isRegression = model.m.getModelCategory match {
+      case hex.ModelCategory.Regression => true
+      case hex.ModelCategory.Binomial   => false
```

Comment:
whats a binomial model.

Reply:
Binomial model is H2Os binary classification model.